### PR TITLE
Nikhil/docs sprint

### PR DESCRIPTION
**Pull request by nikhilgarg28**

### CODE REVIEW

File: docs/pages/api-reference/data-types.md [hunk 1]
```diff
@@ -33,13 +33,26 @@ Fennel supports the following data types, expressed as native Python type hints.
...
 </TypesList>
 
 
 Note that types don't auto-typecast. For instance, if something was expected to
 be of type `float` but received an `int`, Fennel will declare that to be an error.
 
 
+**Struct Type**
+
+Fennel natively supports 'struct' type to represent a bag of fixed typed fields.
```
Comment (aditya-nambiar) [inline docs/pages/api-reference/data-types.md:49]: is 'struct' better than bag ?

File: docs/pages/architecture/cost-optimizations.md [hunk 2]
```diff
@@ -7,20 +7,21 @@ status: 'published'
...
 Production feature engineering systems deal with lots of data at low latencies
-and hence can be very costly on cloud infrastructures. Fennel goes an extra
-mile in trying to keep cloud costs by squeezing as much out of hardware as 
-possible. This is accomplished by a long tail of cost optimizations -- many of
-which are too low ROI for each team to invest in individually but become feasible
-due to economies of scale enjoyed by Fennel.
+and hence can be very costly in terms of cloud infrastructure costs. Fennel 
+goes an extra mile in trying to keep cloud costs low by squeezing as much out 
+of cloud hardware as possible. This is accomplished by a long tail of cost 
+optimizations -- many of which are too low ROI for each team to invest in
```
Comment (aditya-nambiar) [inline docs/pages/architecture/cost-optimizations.md:13]: too low "an" ROI or too low of an ROI. R is consonant but I believe "an" is the right article

File: docs/pages/data-quality/feature-drift.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+features. It does so by measuring the mean and the standard deviation of the
+extracted feature values.
+
+Changes in the probability distribution of a feature are usually indicative of 
+one of the three scenarios:
+
+1. Organic drift over time
+2. Major changes to feature definitions
+3. Newly introduced bugs
+
```
Comment (aditya-nambiar) [inline docs/pages/data-quality/feature-drift.md:19]: Feedback cycle / Sudden changes in user behavior ?

**Review by aditya-nambiar — APPROVED** [review]

File: docs/examples/api-reference/data-types.py [hunk 4]
```diff
@@ -4,13 +4,33 @@
...
-from fennel.lib.schema import oneof, between, regex
+from fennel.lib.schema import oneof, between, regex, struct
 from fennel.sources import source, Webhook
 from fennel.test_lib import mock
 
 webhook = Webhook(name="fennel_webhook")
 
 
+# docsnip struct_type
+@struct  # like dataclass but verifies that all fields are valid Fennel types
```
[the remark this answers is not in the document]
Comment (nikhilgarg28) [reply docs/examples/api-reference/data-types.py:15]: I wanted the comment to still fit in 80 chars so made it slightly less verbose.

File: docs/pages/api-reference/data-types.md (hunk 1 above)
Comment (nikhilgarg28) [reply docs/pages/api-reference/data-types.md:49]: I think so - most languages have a datatype for struct but almost no main stream languages that I know of have a 'bag' data type. So using `struct` brings that immediate mental model.

**Pull request merged**